### PR TITLE
New query parameter validator component

### PR DESCRIPTION
**Pull request by aegypius**

### CODE REVIEW

File: .commitlintrc [hunk 1]
```diff
@@ -16,6 +16,7 @@
         "jsonapi",
         "graphql",
         "openapi",
+        "query-parameter-validator",
```
Comment (soyuka) [inline .commitlintrc]:
```suggestion
        "queryparametervalidator",
```

Comment (soyuka) [reply .commitlintrc]: I think we want commit scopes to be as small as possible

File: src/QueryParameterValidator/QueryParameterValidator.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\QueryParameterValidator;
+
+use ApiPlatform\Api\FilterLocatorTrait;
```
Comment (aegypius) [inline src/QueryParameterValidator/QueryParameterValidator.php]: This trait is deprecated, find a way to deal with the logic another way

File: src/QueryParameterValidator/QueryParameterValidator.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+use ApiPlatform\Exception\FilterValidationException;
```
Comment (aegypius) [inline src/QueryParameterValidator/QueryParameterValidator.php]: @soyuka This is not the right exception, the scope is not binded to the filter but to the query ? how to deal with it ?

File: src/QueryParameterValidator/QueryParameterValidator.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+use ApiPlatform\QueryParameterValidator\Validator\ArrayItems;
+use ApiPlatform\QueryParameterValidator\Validator\Bounds;
+use ApiPlatform\QueryParameterValidator\Validator\Enum;
+use ApiPlatform\QueryParameterValidator\Validator\Length;
+use ApiPlatform\QueryParameterValidator\Validator\MultipleOf;
+use ApiPlatform\QueryParameterValidator\Validator\Pattern;
+use ApiPlatform\QueryParameterValidator\Validator\Required;
+use Psr\Container\ContainerInterface;
```
Comment (aegypius) [inline src/QueryParameterValidator/QueryParameterValidator.php]: Needs a psr/container requirement

File: src/Metadata/Exception/ValidationException.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+declare(strict_types=1);
+
+namespace ApiPlatform\Metadata\Exception;
+
+/**
+ * Filter validation exception.
+ *
+ * @author Julien DENIAU <julien.deniau@gmail.com>
+ */
+final class ValidationException extends \Exception implements ExceptionInterface, \Stringable
```
Comment (soyuka) [inline src/Metadata/Exception/ValidationException.php]: shouldn't that be inside QueryParameterValidator?

**Comment by aegypius** [comment]
Merged upstream as https://github.com/api-platform/core/pull/6080

**Pull request closed without merging**